src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs

```diff
@@ -242,24 +239,39 @@ public override async Task StartProcessingMessagesAsync(CancellationToken cancel
             }
 
             await base.StartProcessingMessagesAsync(cancellationToken);
-            
-            if (_messageProcessor is null)
-            {
-                _messageProcessor = await Settings.CreateMessageProcessorAsync();
-            }
-
-            Namespace = _messageProcessor.FullyQualifiedNamespace;
 
             /* TODO: we can't support Azure Service Bus plug-ins yet because the new Azure SDK doesn't yet support this:
-                     https://github.com/arcus-azure/arcus.messaging/issues/176 */
-
-            RegisterClientInformation(JobId, _messageProcessor.EntityPath);
+                   https://github.com/arcus-azure/arcus.messaging/issues/176 */
 
-            Logger.LogTrace("Starting Azure Service Bus {EntityType} message pump '{JobId}' on entity path '{EntityPath}' in namespace '{Namespace}'", Settings.ServiceBusEntity, JobId, EntityPath, Namespace);
-            _messageProcessor.ProcessErrorAsync += ProcessErrorAsync;
-            _messageProcessor.ProcessMessageAsync += ProcessMessageAsync;
-            await _messageProcessor.StartProcessingAsync(cancellationToken);
             Logger.LogInformation("Azure Service Bus {EntityType} message pump '{JobId}' on entity path '{EntityPath}' in namespace '{Namespace}' started", Settings.ServiceBusEntity, JobId, EntityPath, Namespace);
+
+            Namespace = _messageReceiver.FullyQualifiedNamespace;
+            RegisterClientInformation(JobId, _messageReceiver.EntityPath);
+
+            _receiveMessagesCancellation = new CancellationTokenSource();
+            while (CircuitState is MessagePumpCircuitState.Closed 
+                   && !_messageReceiver.IsClosed 
+                   && !_receiveMessagesCancellation.IsCancellationRequested)
+            {
+                try
+                {
+                    IReadOnlyList<ServiceBusReceivedMessage> messages = 
+                        await _messageReceiver.ReceiveMessagesAsync(Settings.Options.MaxConcurrentCalls, cancellationToken: _receiveMessagesCancellation.Token);
+
+                    await Task.WhenAll(messages.Select(msg => ProcessMessageAsync(msg, cancellationToken)));
+                }
+                catch (Exception exception) when (exception is TaskCanceledException or OperationCanceledException or ObjectDisposedException)
+                {
+                    IsStarted = false;
+
+                    Logger.LogTrace("Azure Service Bus {EntityType} message pump '{JobId}' on entity path '{EntityPath}' in '{Namespace}' was cancelled", Settings.ServiceBusEntity, JobId, EntityPath, Namespace);
+                    return;
+                }
+                catch (Exception exception)
+                {
+                    await ProcessErrorAsync(exception, cancellationToken);
```

Comment:
This is a bit problematic I guess.
I've been playing around with this in a demo-project and I simulate an error that occurs during processing by throwing an exception.
Normally, the message that could not be processed must be re-delivered to the queue (or topic) so that it can be retried in a next iteration.   Eventually, when the message has been re-delivered a number of time, ServiceBus will deadletter the message.

So, in my demo-app I expect to see some messages in the deadletter queue, but there are none present.  This means that failed messages are being removed from the queue/topic because I believe the 'Complete' method is called as the exception that is thrown in the message-handler is eventually caught here and 'eaten'.
In case of an unhandled exception, we must call the Abandon functionality for the message that could not be processed.

Reply:
Ah, abondoning was indeed something that the processor did for us, but we are now responsible with the receiver. Did not noitice. Will add/test this.

Reply:
In case of 'exception swallowing', the messages are auto-abandonned when they does not seem to be processed by the receiver (delivery count is increased). That I got verified in the tests. I added the auto-dead-letter in the message receiver when no message handler was able to process the message.

Reply:
Related to this, I'm wondering if we even have a scenario where the `AutoComplete` option would be set to `false` ?
Is that something that we should continue to support ?  (Breaking change, and also outside the scope of this PR, but just something I was thinking about).

Reply:
If you wanted to complete the message yourself with one of the message handlers (fallback or general), I would think. It gives a bit more control.

Reply:
> In case of 'exception swallowing', the messages are auto-abandonned when they does not seem to be processed by the receiver (delivery count is increased). That I got verified in the tests. I added the auto-dead-letter in the message receiver when no message handler was able to process the message.

That's strange, I've been playing a bit with the latest version (2.0.0-PR-444-20241007.5) and I intentionally throw an exception in the message-handler that is not being caught.  The messages end up in the deadletter queue, however, their Delivery Count property remains 0, which is not wat I expected

![image](https://github.com/user-attachments/assets/6ee85150-2106-4769-a256-66ebd871aa4d)

I expected that the DeliveryCount would have been set to 10 and ServiceBus should've assigned a deadletter reason saying that the max delivercount has been reached.

Reply:
Did some further digging; I throw an exception in the MessageHandler which is not being caught.
I would expect that the servicebus message is abandoned and is being returned to the (in this case) queue .  That would cause that the message can be picked up again, and the message-handler would process the message again.
However, this doesn't happen.  The message-handler receives the message for the first time, I throw an exception and the message is being deadlettered right away, although my 'application code' doesn't contain any statement that would deadletter the message.

Reply:
Yes that that is expected behavior, actually. We already talked about this - we can't know if a message handler throws an exception because it can't handle it, or because the backend system is failing, or the implementation is wrong... We only know that it can't match it, so we go to the next message handler and if there is none, then we dead-letter.

Reply:
We should not deadletter immediately.  The message must be abandoned which means it's delivered back to the queue or topic.
The message will be received again, and will be routed once again through the message-handlers.   If we have a matching message-handler, it will process the message once again (you can see it as the message processing being retried).  And this goes on as many times until the message gets finally succesfully processed or until the max amount of deliveries has been reached.  In the latter case, the message will be deadlettered.
What I've described is also the behaviour that currently exists in Arcus.Messaging 2.0.0

Reply:
But without the receiver, then. Hmm, okay, then we need to investigate a bit more on this on how to verify this - will be back on this Monday.

Reply:
It looks like the deadlettering is now working as expected, however, the circuitbreaker doesn't seem to work anymore.  (I see that there are also some tests that are failing ? )

I have created a small test project that consumes messages from a queue and calls an API.  When the API is unavailable, the circuitbreaker is triggered.  
I can see that the messagepump stops (which is expected).  After a while, I can see the circuitbreaker transitioning into a half-open state, as a new message is received.    When the API is available and the message can be processed as expected, I'd expect that the circuitbreaker goes into the closed state, and the messagepump starts retrieving messages from the queue at max configured speed.  However, this doesn't happen.  Moreover, it seems that the MessagePump isn't retrieving anything at all.  Messages remain in the queue.
Even if `ResumeMessageProcessingAsync` is called by the MessageHandler, nothing seems to happen.

There are some failing tests so that might be the indication that there's something going wrong ?

Reply:
Ok, so some additional testing is required - this was actually done better in another PR where we already worked with the new Dev tenant, but we'll have to check here first, I guess.
Currently, not being able to simulate the failure as the tests show a message pump that received messages even after a circuit breaker hapenned. Will need some time to investigate how to come to the same situation.